### ISSUE TITLE
Significant performance gap when training on KITTI from scratch

### ISSUE THREAD

**Issue by AMAZARA5H1** [WHU-USI3DV/CoFiI2P#26]
Dear Authors,
Thank you for your open-source contribution. I am writing to ask for help regarding the reproduction of the CoFiI2P results on the KITTI dataset.
I strictly followed the instructions in your README.md to train the model from scratch, but the final performance is significantly lower than the results reported in your paper.

1. Reproduction Steps
Data: I downloaded the pre-processed KITTI data directly from the link provided in your README.
Code: I used the official code (latest commit) without any modification to the network architecture.
Training Command: I executed the training script using the default settings as specified in the code/README:
Evaluation: Evaluated using mode_epoch_24.t7.

2. Hardware Environment
- GPU: NVIDIA RTX 4090
- CPU: Intel Xeon w5-3435X
- OS: Ubuntu 22.04

3. Results Comparison Despite having sufficient hardware and following the default configuration, the model does not seem to converge to the paper's metrics within 25 epochs.

<img width="950" height="378" alt="Image" src="https://github.com/user-attachments/assets/f144fb0d-cb3a-443c-85d8-b50c0cbadea9" />

4. My Questions Since I used the default configuration (Batch size=1, Epoch=25):
- Is 25 epochs typically sufficient for training from scratch on KITTI, or is this default setting intended for fine-tuning/demo purposes?
- Should I increase the number of epochs (e.g., to 50 or 100) or adjust the learning rate schedule when training from scratch?
- Could there be any implicit settings not mentioned in the README that are critical for achieving the reported accuracy?

I would appreciate any guidance on how to bridge this gap. Thank you!

**Comment by Wow-Song** [comment]
Thanks for the clarification.
I would like to add that I also reproduced similar results to the issue reported above — the training runs successfully, but the performance is significantly lower than that reported in the paper.

Hardware Environment:
GPU：RTX 2080 Ti
CPU：Intel(R) Xeon(R) Platinum 8255C
OS：ubuntu20.04
python：3.8.10

Looking forward to the Docker image. Thank you for your effort!
<img width="589" height="279" alt="Image" src="https://github.com/user-attachments/assets/c30c51bc-55fa-4893-9007-cb3ca27102a3" />

> We have already trained and tested on several devices. We plan to attempt to reproduce the issue in early January and subsequently upload a Docker image for training. Thank you for your question.

**Comment by Wow-Song** [comment]
> > Thanks for the clarification. I would like to add that I also reproduced similar results to the issue reported above — the training runs successfully, but the performance is significantly lower than that reported in the paper.
> > Hardware Environment: GPU：RTX 2080 Ti CPU：Intel(R) Xeon(R) Platinum 8255C OS：ubuntu20.04 python：3.8.10
> > Looking forward to the Docker image. Thank you for your effort! <img alt="Image" width="589" height="279" src="https://private-user-images.githubusercontent.com/248382216/530171112-c30c51bc-55fa-4893-9007-cb3ca27102a3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY2NjQ4NDIsIm5iZiI6MTc2NjY2NDU0MiwicGF0aCI6Ii8yNDgzODIyMTYvNTMwMTcxMTEyLWMzMGM1MWJjLTU1ZmEtNDg5My05MDA3LWNiM2NhMjcxMDJhMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNVQxMjA5MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNjA5ZWUwYjM4YjI3YTgyNzkyZThlMzIxOGVlMWFkNzkyY2Q3NjAyZDdiNmQyYTk0YjMxMWU0MDVmZTUyYjhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QYdtPpOGOVwu8B_YqAOK6WA-49GsYGBZYk6cw6c4Dn8">
> > > We have already trained and tested on several devices. We plan to attempt to reproduce the issue in early January and subsequently upload a Docker image for training. Thank you for your question.
> 
> Thank you for your comment. We are unsure whether there might be an issue with the data; do the results obtained using the pre-trained model align with those we reported?

The results obtained using the pre-trained model are consistent with those reported,as shown in the figure.

<img width="798" height="420" alt="Image" src="https://github.com/user-attachments/assets/a935a0d8-488e-4310-989c-4456133b90e1" />

However, the effect of the re-training did not meet the requirements.Evaluated using mode_epoch_24.t7.
The final log output is as follows.

`2025-12-24 11:51:30,543 - __main__ - INFO - train-24-1020110, loss: 6.618628, loss_desc: 2.271791, loss_coarse: 0.007583, loss_fine: 4.339254

2025-12-24 11:51:36,037 - __main__ - INFO - train-24-1020120, loss: 6.587367, loss_desc: 2.261764, loss_coarse: 0.000465, loss_fine: 4.325138

2025-12-24 11:51:41,614 - __main__ - INFO - train-24-1020130, loss: 6.482825, loss_desc: 2.241683, loss_coarse: 0.013873, loss_fine: 4.227270

2025-12-24 11:51:47,013 - __main__ - INFO - train-24-1020140, loss: 6.732367, loss_desc: 2.264800, loss_coarse: 0.002211, loss_fine: 4.465356

2025-12-24 11:51:52,438 - __main__ - INFO - train-24-1020150, loss: 6.778427, loss_desc: 2.425799, loss_coarse: 0.000134, loss_fine: 4.352494

2025-12-24 11:51:57,837 - __main__ - INFO - train-24-1020160, loss: 6.555468, loss_desc: 2.279044, loss_coarse: 0.000001, loss_fine: 4.276423

2025-12-24 11:52:03,242 - __main__ - INFO - train-24-1020170, loss: 6.512969, loss_desc: 2.301415, loss_coarse: 0.012676, loss_fine: 4.198877

2025-12-24 11:52:08,538 - __main__ - INFO - train-24-1020180, loss: 6.692964, loss_desc: 2.318229, loss_coarse: 0.000114, loss_fine: 4.374621

2025-12-24 11:52:13,844 - __main__ - INFO - train-24-1020190, loss: 6.515771, loss_desc: 2.278768, loss_coarse: 0.001912, loss_fine: 4.235090

2025-12-24 11:52:19,137 - __main__ - INFO - train-24-1020200, loss: 6.578541, loss_desc: 2.322017, loss_coarse: 0.030835, loss_fine: 4.225689

2025-12-24 11:52:30,900 - __main__ - INFO - acc: top5 0.08506944 0.14583334 0.19444443 0.22395831 0.25694445

2025-12-24 11:52:36,040 - __main__ - INFO - train-24-1020210, loss: 6.815390, loss_desc: 2.387784, loss_coarse: 0.003248, loss_fine: 4.424357

2025-12-24 11:52:41,636 - __main__ - INFO - train-24-1020220, loss: 6.567030, loss_desc: 2.311119, loss_coarse: 0.007810, loss_fine: 4.248102

2025-12-24 11:52:47,034 - __main__ - INFO - train-24-1020230, loss: 6.656217, loss_desc: 2.377336, loss_coarse: 0.000562, loss_fine: 4.278319

2025-12-24 11:52:52,545 - __main__ - INFO - train-24-1020240, loss: 6.554456, loss_desc: 2.269868, loss_coarse: 0.031204, loss_fine: 4.253385

2025-12-24 11:52:58,114 - __main__ - INFO - train-24-1020250, loss: 6.767127, loss_desc: 2.356939, loss_coarse: 0.001768, loss_fine: 4.408420

2025-12-24 11:53:03,544 - __main__ - INFO - train-24-1020260, loss: 6.765138, loss_desc: 2.375539, loss_coarse: 0.007660, loss_fine: 4.381939

2025-12-24 11:53:09,014 - __main__ - INFO - train-24-1020270, loss: 6.764698, loss_desc: 2.319659, loss_coarse: 0.001887, loss_fine: 4.443151

2025-12-24 11:53:14,436 - __main__ - INFO - train-24-1020280, loss: 6.606939, loss_desc: 2.304106, loss_coarse: 0.003187, loss_fine: 4.299646

2025-12-24 11:53:19,838 - __main__ - INFO - train-24-1020290, loss: 6.748415, loss_desc: 2.259939, loss_coarse: 0.006330, loss_fine: 4.482146

2025-12-24 11:53:25,031 - __main__ - INFO - train-24-1020300, loss: 6.680527, loss_desc: 2.270117, loss_coarse: 0.000109, loss_fine: 4.410301

2025-12-24 11:53:36,498 - __main__ - INFO - acc: top5 0.08506944 0.14756945 0.19791664 0.2326389 0.26041666

2025-12-24 11:53:41,721 - __main__ - INFO - train-24-1020310, loss: 6.792307, loss_desc: 2.313724, loss_coarse: 0.000670, loss_fine: 4.477913

2025-12-24 11:53:46,839 - __main__ - INFO - train-24-1020320, loss: 6.592173, loss_desc: 2.346738, loss_coarse: 0.004454, loss_fine: 4.240981

2025-12-24 11:53:52,038 - __main__ - INFO - train-24-1020330, loss: 6.475137, loss_desc: 2.232943, loss_coarse: 0.006919, loss_fine: 4.235275

2025-12-24 11:53:57,243 - __main__ - INFO - train-24-1020340, loss: 6.715595, loss_desc: 2.331777, loss_coarse: 0.026167, loss_fine: 4.357651

2025-12-24 11:54:02,438 - __main__ - INFO - train-24-1020350, loss: 6.489400, loss_desc: 2.274071, loss_coarse: 0.002053, loss_fine: 4.213276

2025-12-24 11:54:07,838 - __main__ - INFO - train-24-1020360, loss: 6.536842, loss_desc: 2.313475, loss_coarse: 0.001109, loss_fine: 4.222259

2025-12-24 11:54:13,136 - __main__ - INFO - train-24-1020370, loss: 6.697450, loss_desc: 2.262609, loss_coarse: 0.028426, loss_fine: 4.406415

2025-12-24 11:54:18,439 - __main__ - INFO - train-24-1020380, loss: 6.491575, loss_desc: 2.259993, loss_coarse: 0.001979, loss_fine: 4.229603

2025-12-24 11:54:23,536 - __main__ - INFO - train-24-1020390, loss: 6.853717, loss_desc: 2.365267, loss_coarse: 0.003324, loss_fine: 4.485126

2025-12-24 11:54:28,844 - __main__ - INFO - train-24-1020400, loss: 6.685488, loss_desc: 2.338259, loss_coarse: 0.001874, loss_fine: 4.345355

2025-12-24 11:54:41,057 - __main__ - INFO - acc: top5 0.083333336 0.14756945 0.19618054 0.22395833 0.25520834

2025-12-24 11:54:46,339 - __main__ - INFO - train-24-1020410, loss: 6.701672, loss_desc: 2.297564, loss_coarse: 0.001156, loss_fine: 4.402951

2025-12-24 11:54:51,738 - __main__ - INFO - train-24-1020420, loss: 6.730059, loss_desc: 2.326975, loss_coarse: 0.001097, loss_fine: 4.401987

2025-12-24 11:54:56,836 - __main__ - INFO - train-24-1020430, loss: 7.056482, loss_desc: 2.466043, loss_coarse: 0.000004, loss_fine: 4.590435

2025-12-24 11:55:01,278 - __main__ - INFO - train-24-1020440, loss: 6.832616, loss_desc: 2.417321, loss_coarse: 0.011372, loss_fine: 4.403922

2025-12-24 11:55:04,703 - __main__ - INFO - train-24-1020450, loss: 6.846061, loss_desc: 2.269501, loss_coarse: 0.007809, loss_fine: 4.568751`

**Comment by blossomzzj** [comment]
I’m facing the exact same issue! Couldn’t replicate the paper’s performance either, and sharing the Docker environment and dataset would be incredibly helpful.

**Comment by Wow-Song** [comment]
I have identified the possible cause of the problem. Because random translations were added in the x-y plane during the evaluation, the results were quite different. When I removed the random translation, everything returned to normal.

Hope this can be of help to you all.
